modules/aruco/samples/calibrate_camera_charuco.cpp

```diff
@@ -57,8 +57,8 @@ const char* about =
 const char* keys  =
         "{w        |       | Number of squares in X direction }"
         "{h        |       | Number of squares in Y direction }"
-        "{sl       |       | Square side lenght (in pixels) }"
-        "{ml       |       | Marker side lenght (in pixels) }"
+        "{sl       |       | Square side length (in meters) }"
+        "{ml       |       | Marker side length (in meters) }"
```

Comment:
Indeed, to calibrate intrinsic parameters, the unit you use is completely irrelevant (you can use mm, m, inches or anything). This is simply because, in the pinhole model, what you use is the ratio X/Z and Y/Z respectively for x and y. So as far as all the points are in the same unit, it is not relevant. In this concrete sample, as far as -sl and -ml are in the same unit, it should work ok.
On the contrary, when you are estimating the pose (the extrinsic parameters), the unit you use is relevant since the translation vector will be estimated in the same unit (for instance in the detect_markers or detect_board samples).

However, to simplify this for the users and avoid confusions/errors, we just decided to "force" them to use always the same unit by specifying "(in meters)" in all the samples. It seems that, in a later PR, someone changed it to "(in pixels)" by mistake and I missed that in the revision.